wallet/core/src/account/pskb.rs

```diff
@@ -0,0 +1,346 @@
+                                let public_key = PublicKey::from_keypair(&keypair);
+
+                                return Some(Ok(SignInputOk {
+                                    signature: Signature::Schnorr(signature),
+                                    pub_key: public_key,
+                                    key_source: None,
+                                }));
+                            }
+                        }
+                        None
+                    })
+                    .collect::<Result<Vec<_>, _>>()
+            })
+            .unwrap();
+        return Ok(signed_pskt);
+    }
+
+    Ok(signer)
+}
+
+pub async fn bundle_from_pskt_generator(generator: PSKTGenerator) -> Result<Bundle, Error> {
+    let mut bundle = Bundle::new();
+    let mut stream = generator.stream();
+
+    while let Some(pskt_result) = stream.next().await {
+        match pskt_result {
+            Ok(pskt) => bundle.add_pskt(pskt),
+            Err(e) => return Err(e),
+        }
+    }
+
+    Ok(bundle)
+}
+
+pub async fn pskb_signer(bundle: &Bundle, signer: Arc<PSKBSigner>, network_id: NetworkId) -> Result<Bundle, Error> {
+    let mut signed_bundle = Bundle::new();
+    let mut reused_values = SigHashReusedValues::new();
+
+    for pskt_inner in bundle.inner_list.clone() {
+        let pskt: PSKT<Signer> = PSKT::from(pskt_inner);
+
+        let addresses: Vec<Address> = pskt.inputs.iter()
+            .filter_map(|input| input.utxo_entry.as_ref()) // Filter out None and get a reference to UtxoEntry if it exists
+            .map(|utxo_entry| extract_script_pub_key_address(&utxo_entry.script_public_key.clone(), network_id.into()).unwrap()) // Clone the ScriptPublicKey
+            .collect();
+
+        signer.ingest(addresses.as_ref())?;
+
+        // Define the signing function
+        let mut sign = |signer_pskt: PSKT<Signer>| {
+            signer_pskt
+                .pass_signature_sync(|tx, sighash| -> Result<Vec<SignInputOk>, String> {
+                    tx.tx
+                        .inputs
+                        .iter()
+                        .enumerate()
+                        .map(|(idx, _input)| {
+                            let hash = calc_schnorr_signature_hash(&tx.as_verifiable(), idx, sighash[idx], &mut reused_values);
+                            let msg = secp256k1::Message::from_digest_slice(hash.as_bytes().as_slice()).unwrap();
+                            let address = addresses.get(idx).expect("Input indexed address");
+                            let public_key = signer.public_key(address).expect("Public key for input indexed address");
+                            Ok(SignInputOk {
+                                signature: Signature::Schnorr(signer.sign_schnorr(address, msg).unwrap()),
+                                pub_key: public_key,
+                                key_source: None,
+                            })
```

Comment:
is it possible to pass derivation path and fingerprint here?

Reply:
Done!